### PR TITLE
Use explicit memory handling for db and always free

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Always use the span overloads for RocksDb Get and explicitly free rather than using underlying lib's handling

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [x] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [x] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs [hunk 1]
```diff
@@ -372,7 +372,24 @@ public long GetSize()
...
         {
-            return _db.Get(key);
+            Span<byte> data = _db.GetSpan(key);
+            if (data.IsNull())
+            {
+                return null;
+            }
+            try
+            {
+                if (data.Length == 0)
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:382]: Maybe if length is 0, it still need to free it?

Comment (benaadams) [reply src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs:382]: Will still free it as happens in the `finally` block

**Review by asdacap — APPROVED** [review]
Does it make a difference?

**Comment by benaadams** [comment]
> Does it make a difference?

Nope 😢 Thought was worth a try though

**Comment by LukaszRozmej** [comment]
If it doesn't make a difference lets close it?

**Pull request closed without merging**